### PR TITLE
merge user and user_detail table into users table

### LINKED ISSUE #3644: Merge user and user_detail table and maintain data integrity
**I'm submitting a ...**  (check one with "x")
- [x] bug report
- [ ] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-orga-server

**Current behavior:**

Currently the user and user_detail have on 1-1 relationship with user_id acting as the primary key in both tables.

**Expected behavior:**

Should be a single table

**Steps to reproduce:**

Create the database.

### PR DESCRIPTION
**Pull request by shubham-padia**
maintains data integrity

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
merge user and user_detail table

#### Changes proposed in this pull request:

- merge user and user_detail table
- relevant refactors
-

Fixes #3644

### CODE REVIEW

File: app/old_api/old_tests/__init__.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import unittest
+mport unittest
```
Comment (SaptakS) [inline app/old_api/old_tests/__init__.py]: Please check this.

Comment (shubham-padia) [inline app/old_api/old_tests/__init__.py]: Thanks !!

File: migrations/versions/9105f9e64cdc_.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    op.create_foreign_key(u'speaker_user_id_fkey', 'speaker', 'users', ['user_id'], ['id'], ondelete='SET NULL')
+    op.drop_constraint(u'transaction_user_id_fkey', 'transaction', type_='foreignkey')
+    op.create_foreign_key(u'transaction_user_id_fkey', 'transaction', 'users', ['user_id'], ['id'])
+    op.drop_constraint(u'user_system_role_user_id_fkey', 'user_system_role', type_='foreignkey')
+    op.create_foreign_key(u'user_system_role_user_id_fkey', 'user_system_role', 'users', ['user_id'], ['id'], ondelete='CASCADE')
+    op.drop_constraint(u'users_events_roles_user_id_fkey', 'users_events_roles', type_='foreignkey')
+    op.create_foreign_key(u'users_events_roles_user_id_fkey', 'users_events_roles', 'users', ['user_id'], ['id'], ondelete='CASCADE')
+    op.execute("INSERT INTO users SELECT temp_user.* , user_detail.firstname, user_detail.lastname, user_detail.details, user_detail.contact, user_detail.facebook, user_detail.twitter, user_detail.instagram, user_detail.google, user_detail.avatar_uploaded, user_detail.thumbnail, user_detail.small, user_detail.icon FROM temp_user INNER JOIN user_detail ON temp_user.id = user_detail.user_id",
```
Comment (niranjan94) [inline migrations/versions/9105f9e64cdc_.py]: Cool 👍

**Comment by niranjan94** [comment]
@shubham-padia error when the migrations ran
```
INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade 16caacf7c5b0 -> 9105f9e64cdc, empty message
Traceback (most recent call last):
  File "manage.py", line 98, in <module>
    manager.run()
  File "/usr/local/lib/python2.7/site-packages/flask_script/__init__.py", line 412, in run
    result = self.handle(sys.argv[0], sys.argv[1:])
  File "/usr/local/lib/python2.7/site-packages/flask_script/__init__.py", line 383, in handle
    res = handle(*args, **config)
  File "/usr/local/lib/python2.7/site-packages/flask_script/commands.py", line 216, in __call__
    return self.run(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/flask_migrate/__init__.py", line 244, in upgrade
    command.upgrade(config, revision, sql=sql, tag=tag)
  File "/usr/local/lib/python2.7/site-packages/alembic/command.py", line 254, in upgrade
    script.run_env()
  File "/usr/local/lib/python2.7/site-packages/alembic/script/base.py", line 421, in run_env
    util.load_python_file(self.dir, 'env.py')
  File "/usr/local/lib/python2.7/site-packages/alembic/util/pyfiles.py", line 93, in load_python_file
    module = load_module_py(module_id, path)
  File "/usr/local/lib/python2.7/site-packages/alembic/util/compat.py", line 75, in load_module_py
    mod = imp.load_source(module_id, path, fp)
  File "migrations/env.py", line 89, in <module>
    run_migrations_online()
  File "migrations/env.py", line 82, in run_migrations_online
    context.run_migrations()
  File "<string>", line 8, in run_migrations
  File "/usr/local/lib/python2.7/site-packages/alembic/runtime/environment.py", line 817, in run_migrations
    self.get_context().run_migrations(**kw)
  File "/usr/local/lib/python2.7/site-packages/alembic/runtime/migration.py", line 329, in run_migrations
    step.migration_fn(**kw)
  File "/opev/open_event/migrations/versions/9105f9e64cdc_.py", line 59, in upgrade
    op.drop_constraint(u'email_notifications_user_id_fkey', 'email_notifications', type_='foreignkey')
  File "<string>", line 8, in drop_constraint
  File "<string>", line 3, in drop_constraint
  File "/usr/local/lib/python2.7/site-packages/alembic/operations/ops.py", line 148, in drop_constraint
    return operations.invoke(op)
  File "/usr/local/lib/python2.7/site-packages/alembic/operations/base.py", line 318, in invoke
    return fn(self, operation)
  File "/usr/local/lib/python2.7/site-packages/alembic/operations/toimpl.py", line 146, in drop_constraint
    schema=operation.schema,
  File "/usr/local/lib/python2.7/site-packages/alembic/ddl/impl.py", line 183, in drop_constraint
    self._exec(schema.DropConstraint(const))
  File "/usr/local/lib/python2.7/site-packages/alembic/ddl/impl.py", line 118, in _exec
    return conn.execute(construct, *multiparams, **params)
  File "/usr/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 945, in execute
    return meth(self, multiparams, params)
  File "/usr/local/lib/python2.7/site-packages/sqlalchemy/sql/ddl.py", line 68, in _execute_on_connection
    return connection._execute_ddl(self, multiparams, params)
  File "/usr/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1002, in _execute_ddl
    compiled
  File "/usr/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1189, in _execute_context
    context)
  File "/usr/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1402, in _handle_dbapi_exception
    exc_info
  File "/usr/local/lib/python2.7/site-packages/sqlalchemy/util/compat.py", line 203, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/usr/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1182, in _execute_context
    context)
  File "/usr/local/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 470, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.ProgrammingError: (psycopg2.ProgrammingError) constraint "email_notifications_user_id_fkey" of relation "email_notifications" does not exist
 [SQL: 'ALTER TABLE email_notifications DROP CONSTRAINT email_notifications_user_id_fkey']
```

**Comment by shubham-padia** [comment]
@niranjan94 That is weird. Was this error on the api.eventyay.com deployment? I also created a test database upto migration `16caacf7c5b0`, and then ran this migration. Both upgrades and downgrades are working fine on it.

**Comment by niranjan94** [comment]
Yes. on `api.eventyay.com`

**Comment by shubham-padia** [comment]
The database for the deployment was created fresh when the first deployment for nextgen was done, right ?

**Comment by niranjan94** [comment]
Yes.

**Comment by niranjan94** [comment]
@shubham-padia actually no. The staging database was re-used. To simulate the the switch.

**Comment by shubham-padia** [comment]
@niranjan94 Found the problem, during the schema refactor, looks like renaming "email_notification_user_id_fkey" to ""email_notification_user_id_fkey" was missed during the schema refactor.

**Comment by niranjan94** [comment]
Okay ... Could you make a PR with that ?

**Comment by shubham-padia** [comment]
Yeah, i'm checking for other inconsistencies if they exist, and will make a PR after that.

**Comment by niranjan94** [comment]
Awesome 😄

**Pull request merged**